e2e/nomostest/watch.go

```diff
@@ -121,15 +224,67 @@ func watchForObjectInner(nt *NT, gvk schema.GroupVersionKind, name, namespace st
 			prevErrs = errs
 			prevObj = cObj
 		case watch.Error:
-			return errors.Errorf("watch error: %+v", event)
+			return errors.Errorf("%s recieved error event: %#v",
+				errPrefix, event)
 		default:
-			return errors.Errorf("unexpected watch event type %q", event.Type)
+			return errors.Errorf("%s recieved unexpected event: %#v",
+				errPrefix, event)
 		}
 	}
 
 	if len(prevErrs) > 0 {
 		// watch exited with predicate errors
 		return multierr.Combine(prevErrs...)
 	}
-	return errors.Errorf("watch exited before any add/update events were recieved")
+	return errors.Errorf("%s exited before any add/update events were recieved",
+		errPrefix)
+}
+
+// WatchForCurrentStatus watches the object until it reconciles (Current).
+func WatchForCurrentStatus(nt *NT, gvk schema.GroupVersionKind, name, namespace string, opts ...WatchOption) error {
+	return WatchObject(nt, gvk, name, namespace,
+		[]Predicate{StatusEquals(nt, kstatus.CurrentStatus)},
+		opts...)
+}
+
+// WatchForNotFound waits for the passed object to be fully deleted or not found.
+// Returns an error if the object is not deleted within the timeout.
+func WatchForNotFound(nt *NT, gvk schema.GroupVersionKind, name, namespace string, opts ...WatchOption) error {
+	opts = append(opts, watchErrorOnNotFound())
+	err := WatchObject(nt, gvk, name, namespace,
+		[]Predicate{objectNotFoundPredicate},
+		opts...)
+	if err == nil {
+		// Should never happen!
+		panic("WatchForNotFound exited without error or timeout")
```

Comment:
It will happen if `errorOnNotFound` is false and a message will be logged. Do we still need to panic?

Reply:
errorOnNotFound is a private field on a private struct modified by a private function. No caller outside of this package can use it. It's only enabled by this function, and it's enabled by appending to the caller's options, overriding what the caller specified. So it will always be true in this context.

Reply:
ACK